server/controllers/admin/users/index.js

```diff
@@ -81,12 +81,34 @@ function list(req, res, next) {
   const sql = 'SELECT user.id, display_name, user.username, user.deactivated FROM user;';
 
   db.exec(sql)
-    .then((rows) => {
-      res.status(200).json(rows);
+    .then((users) => {
+      return setRoles(users);
+    }).then(users => {
+
+      res.status(200).json(users);
     })
     .catch(next);
 }
 
+async function setRoles(users) {
+  const sql = `
+    SELECT ur.user_id, r.label
+    FROM user_role ur
+    JOIN role r ON r.uuid = ur.role_uuid`;
+
+  const userMap = {};
+  users.forEach(user => {
+    user.roles = '';
+    userMap[user.id] = user;
+  });
+
+  const roles = await db.exec(sql);
```

Comment:
This code might be a bit cleaner by looping through the roles and grouping by user.  Something like this:
```js
const roles = await db.exec(sql);

users.forEach(user => {
  user.roles = roles
    .filter(role => role.user_id === user.id)
    .map(role => role.label)
    .join(', ');
});

return users;
```

Reply:
A benefit of using .join is that it keeps a comma from trailing at the end of each Roles field.

Reply:
FYI, I submitted the suggested update above as a commit. https://github.com/IMA-WorldHealth/bhima-2.X/commit/ed876e8f69c4a8f50552764e2a90135497012ee6